### ISSUE TITLE
Training a pipeline

### ISSUE THREAD

**Issue by giguru** [giguru/converse#4]
In Haystack, now only the one reader is trainable, but you would like to be able to train an entire pipeline. (See method FarmReader.train)
 
How does training a pipeline of multiple retrievers work? As in: how does the backpropagation work in that training process?

- How does the loss transfer from one retriever to the previous retriever?
- A pipeline of retrievers can have non-transformer based pipeline retriever, e.g. DPR followed by a BM25 retriever, then followed by a LTR?
How do you compute the losses of the DPR and the LTR?
